src/encoder.js

```diff
@@ -388,62 +416,77 @@ export default class Encoder extends Extendable {
       target.push(tags.BINARY_);
       this.encodeValue(value.byteLength, target);
       this.pushArrayElements(new Uint8Array(value), target);
-    } else if (Array.isArray(value)) {
-      let numElements = value.length;
+    } else {
+      if (this.remainingDepth !== null) {
+        if (this.remainingDepth === 0) {
+          target.push(tags.EXTENSION);
+          // $FlowFixMe if remainingDepth !== null, depthBoundExtensionPoint is a number
```

Comment:
This is an invariant that you know, but it could be broken by non-local changes. I'd say it's worth doing a test here anyway.